client/gutenberg/editor/types.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Internal dependencies
+ */
+
+/**
+ * Types
+ */
+import * as T from 'types';
+
+export enum EditorActions {
+	GoToAllPosts = 'goToAllPosts',
+	OpenMediaModal = 'openMediaModal',
+	OpenRevisions = 'openRevisions',
+	PreviewPost = 'previewPost',
+	SetDraftId = 'draftIdSet',
+	TrashPost = 'trashPost',
+}
+
+export enum WindowActions {
+	Loaded = 'loaded',
+	ClassicBlockOpenMediaModel = 'classicBlockOpenMediaModal',
+}
+
+export interface MediaModalData {
+	allowedTypes: any;
+	gallery: any;
+	multiple: boolean;
+	value: any;
+}
+
+export type MessageToCalypso =
+	| { action: EditorActions.GoToAllPosts; payload?: { unsavedChanges: boolean } }
```

Comment:
Why lead with the pipe? Looks strange (I wouldn't have imagined it works 🙃 ).

Reply:
hehe. well, I didn't actually. it was the auto formatter that did!

however, I believe that TypeScript has some influence here from some functional languages that do leading punctuation, even…COMMA-FIRST lists 🙃

my guess is that it's because of the symmetry, the same reason we add trailing commas, so things can be added and rearranged without needless diff lines in the commit.

Reply:
I thought the pipe meant, "OR this type", so it seemed odd to not have a first type. ¯\_(ツ)_/¯

Reply:
your intuition was right! but in this case people chose aesthetics over one notion of purity. personally I love having this there on the first item and miss it when a language doesn't have it.

for comparison, the same data type in Elm might end up written like this…

```elm
type MessageToCalypso
  = GoToAllPosts (Maybe Bool)
  | OpenMediaModal MediaModalData
  | OpenRevisions
  | PreviewPost URL
…
```

the symmetry is preserved but it feels inconsistent, especially when rearranging lines

Reply:
> it was the auto formatter that did

Good enough for me! The less formatting choices we need to make (and the less rule exceptions we need to maintain), the better.